File: 217-contains-duplicate/contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # we know that: nums is distinct 
        # does it have neg integers? is nums.length=0?
        # brute force - 2 for loops - sort check for i and i+1
        hash = set()
        for num in nums:
            if num in hash:
                return True
            hash.add(num)

        return False

File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        //1. create a hashset
        Set<Integer> dupes = new HashSet<>();
        //2. run a loop through each nums
        for(int num : nums){
            //3. check if the set contains the num 
            if(dupes.contains(num))
                return true; //duplicate exists
            dupes.add(num);  //otherwise add it to the set
        }
        return false; //duplicate found!
    }
}

//brute force - create two loops and compare values -- not ideal as the tc = O(nxn)
//Optimal way - hashset
